src/aleph/http/client.clj

```diff
@@ -609,6 +614,16 @@
        ([_ ctx]
          (let [ch (.channel ctx)]
            (reset! in (netty/buffered-source ch (constantly 1) 16))
+
+           ;; Start handshake timeout timer
+           (reset! timeout-task
+                   (.in time/*clock*
+                        handshake-timeout
+                        (fn []
+                          ;; if there was no answer after handshake was sent - close channel
+                          (d/error! d (WebSocketHandshakeTimeoutException. "WebSocket handshake timeout"))
+                          (netty/close ctx))))
```

Comment:
We're not on I/O thread here, so we should probably close the channel, not the context. Right? @ztellman